src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -324,6 +326,25 @@ uct_cuda_copy_md_query_attributes(uct_cuda_copy_md_t *md, const void *address,
          * allocated in a context should also allows us to identify
          * virtual/stream-ordered CUDA allocations. */
         mem_info->type = UCS_MEMORY_TYPE_CUDA_MANAGED;
+
+        cu_err =
+            cuMemRangeGetAttribute((void*)&pref_loc, 4,
```

Comment:
`cuMemRangeGetAttribute(&cuda_device, sizeof(cuda_device),`

Reply:
@yosefe The documentation for cuMemRangeGetAttribute states this about using PREFERRED_LOCATION:

"
[CU_MEM_RANGE_ATTRIBUTE_PREFERRED_LOCATION](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__TYPES.html#group__CUDA__TYPES_1gg3754eb90f64ed3c19b4e550d21d124fcc68a2f5771f8e1b83ad29c1d65ab4875): If this attribute is specified, data will be interpreted as a 32-bit integer, and dataSize must be 4. The result returned will be a GPU device id if all pages in the memory range have that GPU as their preferred location, or it will be CU_DEVICE_CPU if all pages in the memory range have the CPU as their preferred location, or it will be CU_DEVICE_INVALID if either all the pages don't have the same preferred location or some of the pages don't have a preferred location at all. Note that the actual location of the pages in the memory range at the time of the query may be different from the preferred location.
"

The change you suggested assumes that `CUdevice` is a 32-bit integer and size is always 4 bytes. This is true for now but isn't forward compatible. Should I still make the change?
```
typedef int CUdevice_v1;                                     /**< CUDA device */
typedef CUdevice_v1 CUdevice;                                /**< CUDA device */
```

Reply:
option1: UCS_STATIC_ASSERT(sizeof(cuda_device) == sizeof(uint32_t))
option2: "uint32_t pref_loc; /* Argument must be a 32-bit integer */"